platform/measurement_utils.cpp

```diff
@@ -49,7 +53,37 @@ string FormatDistanceImpl(Units units, double m, string const & low, string cons
   {
     double const highV = m / highF;
     // For distances of 10.0 high units and over round to a whole number, e.g. 9.98 -> 10, 10.9 -> 11
-    return ToStringPrecision(highV, round(highV * 10) / 10 >= 10.0 ? 0 : 1) + " " + high;
+
+    int precision = round(highV * 10) / 10 >= 10.0 ? 0 : 1;
+
+    string out = ToStringPrecision(highV, precision);
+
+    if (precision)
+    {
+      // Value with decimals. Set locale decimal separator
```

Comment:
Mac, Linux, and maybe Windows may properly format the string using the system's locale. Can we at least mention it here? If you see that it should not be corrected on Mac, for example, then an `#ifdef` may be needed.

Reply:
We also try to use . at the end of comment sentences. ;-)

Reply:
I've added `#ifdef` so the decimal and grouping separators and "manually" replaced only on Android OS.
I've also added a comment explaining this.